tests/hypothesis_strategies/alternative_creation/test_searchspace.py

```diff
@@ -97,3 +101,34 @@ def test_searchspace_creation_from_dataframe(df, parameters, expected):
     else:
         with pytest.raises(expected):
             SearchSpace.from_dataframe(df, parameters)
+
+
+@given(
+    parameters=st.lists(
+        numerical_discrete_parameter(min_value=0.1, max_value=1.0),
+        min_size=1,
+        unique_by=lambda x: x.name,
+    )
+)
+def tests_discrete_space_creation_from_simplex_inner(parameters):
+    """Candidates from a simplex space satisfy the simplex constraint."""
+    total = 1.0
+    tolerance = 1e-6
+    subspace = SubspaceDiscrete.from_simplex(
+        parameters, total=total, boundary_only=False, tolerance=tolerance
+    )
+    assert (subspace.exp_rep.sum(axis=1) <= total + tolerance).all()
+
+
+def tests_discrete_space_creation_from_simplex_boundary():
+    """Candidates from a simplex boundary space satisfy the boundary constraint."""
+    total = 1.0
+    tolerance = 1e-6
+    parameters = [
```

Comment:
Why are these not provided via @given?

Reply:
Just so that we are on the same page: you suggest that these should be randomly generated by hypothesis as well, correct? In principle I have nothing against that, but I think it is really hard to find a setting that does produce **useful** tests if we did that – which was the reason why I used manual parameterization here. The problem is that you can very easily end up with cases where not a single parameter configuration falls into the requested simplex, in which case the test would trivially succeed (since it checks if the returned configurations, which would be empty then, satisfy the constraint). That is, you need to ensure that at least some of the randomly generated parameters can actually combined such taht you exactly hit the simplex boundary.

Example that can easily happen when you randomly generate without placing very precise restrictions:
* Param1: [0.0, 0.2, 0.5]
* Param2: [0.0, 0.1]
* requested total: 1.0

--> nothing lands on the simplex boundary, test succeeds trivially

You see what I mean? Any ideas?

Reply:
To be honest, I would have expected that hypothesis takes care of that. Isn't the whole point of us using it that we do not need to have manual lists at all? I would at least then include an additional test with hypothesis generating the parameters on its own, then we are certain that we test both important edge cases and the cases hypothesis might find.

Reply:
why not let it draw random parameters and extract from them the `total` instead of fixing it here, eg as sum of max of all values

Reply:
Yeah, refactored the stuff a bit and went back a bit closer to what I had at the very beginning (one single parametrized test for both `boundary_only` options). I think the current version should do it's job and satisfies your request